app/controllers/public/loans_controller.rb

```diff
@@ -1,28 +1,34 @@
 class Public::LoansController < Public::PublicController
   include WordpressEmbeddable
+  after_action :verify_authorized
 
   def index
     params[:division] = get_division_from_url
-    @loans = Loan.filter_by_params(params).visible.
-      includes(:cooperative, division: :super_division).
-      paginate(:page => params[:pg], :per_page => 20).
-      order('signing_date DESC')
+    @params = { status: params[:status], pg: params[:pg], country: params[:country] }
+    @loans = policy_scope(Loan.filter_by_params(params).visible.
+          includes(:organization, division: :parent).
+          page(params[:pg]).per(20).
+          order('signing_date DESC'))
+    authorize @loans
     @countries = Country.order(:iso_code).pluck(:iso_code)
 
+
     # Set last loan list URL for 'Back to Loan List' link
     session[:loans_path] = request.fullpath
   end
 
   def show
-    @loan = Loan.status('all').find(params[:id])
+    @loan = Loan.find(params[:id])
     @pictures = @loan.featured_pictures(5) # for slideshow
-    @other_loans = @loan.cooperative.loans.status('all').order("SigningDate DESC") if @loan.cooperative
-    @repayments = @loan.repayments.order('DateDue')
```

Comment:
Repayments will probably come back in the future so I'd prefer to keep this line here, commented out—I think it's referenced in other commented out code

Reply:
It's not a difficult line to rewrite and the order clause won't work, if repayments are coming back we should track that in redmine rather than leaving dead code lying around that will need to be rewritten to function anyway.